### ISSUE TITLE
Selenium tests fail for some reason on Linux/macOS

### ISSUE THREAD

**Issue by Safihre** [sabnzbd/sabnzbd#2932]
### SABnzbd version

develop

### Operating system

Linux/macOS

### Using Docker image

None

### Description

For some reason the CI fails for a while on all Linux pipelines.
I have no clue why. 
Nothing changed except that GitHub updated the Chrome and ChromeDriver versions.
Errors on macOS are different than on Linux.

**Comment by mnightingale** [comment]
I'm not sure if it's the issue, but I noticed `test_download_basic_rar5` failing here:

https://github.com/sabnzbd/sabnzbd/blame/46170ffb3dabbe8ef0d18abbe8a98b5170d74184/tests/testhelper.py#L320

check_value doesn't have a slash at the end, but its asserting for it.

**Comment by Safihre** [comment]
> Any way to pin/manipulate the chromedriver version on github?

I don't think so :(

**Comment by thezoggy** [comment]
> > Any way to pin/manipulate the chromedriver version on github?
> 
> I don't think so :(

https://github.com/marketplace/actions/setup-chromedriver ?

**Comment by Safihre** [comment]
But what about chrome itself? Supposedly the image provided contains matching Chrome and chromedriver.

**Comment by mnightingale** [comment]
You can let Selenium take care of Chrome versions with:

```py
driver_options = ChromeOptions()
driver_options.browser_version = "120"
```

There is no particular reason I've used 120 here, may want to figure out which version it stops workong on.
I'm also not sure if it helps since I still get failures.

It may also be useful to print/log `driver.capabilities["browserVersion"]`.

**Comment by Safihre** [comment]
@mnightingale you managed to get it to run?

I manged to force it to downgrade. Still get a few errors, but those might really be related to recent changes.
It's quite ugly: https://github.com/sabnzbd/sabnzbd/compare/develop...bugfix/ci

**Comment by mnightingale** [comment]
With https://github.com/mnightingale/sabnzbd/commit/ed55d958fccabb96571c20b496ac519e5cf295b0 actually it still got failures https://github.com/mnightingale/sabnzbd/actions/runs/11001209115/job/30545344228 but maybe they are a seperate issue since it's different tests.

It appears to be the same failurs as you got with the uglier fix, and I'm not sure what to make of them.

Selenium has beta feature to let it manage versions and keep driver/browser in sync https://www.selenium.dev/documentation/selenium_manager/

**Comment by Safihre** [comment]
I finally found the problem for the refused-connection problems.
Turns out there was a much deeper problem with `Option.set`, which would ignore the `default` value of an setting when checking if it should call the callback.
So for this specific case it resulted in a forced restart of SABnzbd when saving the Specials page on a new installation. This resulted in the CI failures, because for about 2 seconds the testing-SABnzbd would be down.
